File: Jeong-Yunmi/6/10828.py
# 10828 스택

N = int(input()) # 명령의 수
stack = []
for _ in range(N):
    command = list(map(str, input().split()))
    if command[0] == "push":
        stack.append(int(command[1]))
    elif command[0] == "pop":
        if len(stack) == 0: print(-1)
        else:               print(stack.pop())
    elif command[0] == "size":
        print(len(stack))
    elif command[0] == "empty":
        if len(stack) == 0: print(1)
        else:               print(0)
    elif command[0] == "top":
        if len(stack) == 0: print(-1)
        else:               print(stack[-1])

File: HaesooJung/6/10828.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
	stack<int> dataStack;      // 스택
	int n;             // 명령의 수
	string command;    // 명령어
	int num;           // push 연산 시, 스택에 넣을 정수
	int result = 0;    // 각 명령어의 결과값

	cin >> n;

	for (int i = 0; i < n; i++) {
		cin >> command;

		// push구현
		if (command == "push") {
			cin >> Inputdata;
			dataStack.push(Inputdata);
		}

		// pop구현
		else if (command == "pop") {
			if (dataStack.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = dataStack.top();
				cout << result << endl;
				dataStack.pop();
			}
		}

		// size구현
		else if (command == "size") {
			cout << dataStack.size() << endl;
		}

		// empty구현
		else if (command == "empty") {
			if (dataStack.size() == 0) {
				result = 1;
				cout << result << endl;
			}
			else {
				result = 0;
				cout << result << endl;
			}
		}

		// top구현
		else if (command == "top") {
			if (dataStack.size() == 0) {
				result = -1;
				cout << result << endl;
			}
			else {
				result = dataStack.top();
				cout << result << endl;
			}
		}
	}

	return 0;
}